vme/src/db.cpp

```diff
@@ -766,60 +766,42 @@ unit_data *read_unit_string(CByteBuffer *pBuf, int type, int len, const char *wh
     if (!str_is_empty(name))
     {
         snprintf(tmpbuf, sizeof(tmpbuf), "%s@%s", name, zone);
-        UNIT_KEY(u) = str_dup(tmpbuf);
+        u->setKey(str_dup(tmpbuf));
     }
     else
     {
-        UNIT_KEY(u) = nullptr;
+        u->setKey(nullptr);
     }
 
-    if (unit_version < 46)
-    {
-        UNIT_MANIPULATE(u) = pBuf->ReadU16(&g_nCorrupt);
-    }
-    else
-    {
-        UNIT_MANIPULATE(u) = pBuf->ReadU32(&g_nCorrupt);
-    }
-
-    UNIT_FLAGS(u) = pBuf->ReadU16(&g_nCorrupt);
-    UNIT_BASE_WEIGHT(u) = pBuf->ReadS16(&g_nCorrupt);
-    UNIT_WEIGHT(u) = pBuf->ReadS16(&g_nCorrupt);
-    UNIT_CAPACITY(u) = pBuf->ReadS16(&g_nCorrupt);
+    u->readManipulateFrom(*pBuf, unit_version, g_nCorrupt);
```

Comment:
I want them gone too, should be a single call for the class to read itself from storage.

I actually wanted to start a discussion about the versioning of the data on discord - I'll start a thread. My general thought was that the vme code should support only 2 versions - current and previous. Then have a utility function to bring datafiles/object up-to-date. I see no reason to write older versions of objects

Reply:
What I'm worried about with these calls is if we mess up something. So the less we rock the code the easier it is to see that it should hopefully be a beniegn change. So doing this:

u32 = pBuf->ReadU32(&g_nCorrupt);
u->setManipulate(u32);

instead of u->readFromManipulate();

is closer to the original
--
So here are a few reasons why having two versions won't be easy. :-) We have a lot of old players saved. Some of them 10-15 years old. They all still load properly. They're on ancient versions. So to deal with that we'd need to build convert scripts and run them regularly. Both on player files but also on all DIL storage in lib/

Secondly, Valheru runs the same engine, and if they want to upgrade they're also on very old versions.

Reply:
So for example I'd prefer this:

    s8 = pBuf->ReadS8(&g_nCorrupt);
    u->SetLights(s8);
    s8 = pBuf->ReadS8(&g_nCorrupt);
    u->SetBright(s8);
    s8 = pBuf->ReadS8(&g_nCorrupt);
    u->SetIllum(s8);

Instead of:

    u->readNumberOfActiveLightSourcesFrom(*pBuf, g_nCorrupt);
    u->readLightOutputFrom(*pBuf, g_nCorrupt);
    u->readTransparentLightOutputFrom(*pBuf, g_nCorrupt);

Reply:
Done.